### PR TITLE
Use collab permissions endpoint to check collab status

### LINKED ISSUE #963: Manage not available when part of organisation
I’m a member of MessageKit org but can’t manage any of the issues/prs even though I should be able to!

<details>
<summary>Bug Report Dump (Auto-generated)</summary>
<pre>
Version 1.13.0 (1190)
Device: iPhone 6s Plus (iOS 11.2)
TestFlight: true
</pre>
</details>

### PR DESCRIPTION
**Pull request by rnystrom**
Fixes #963

Sending out a TF build from this branch

### CODE REVIEW

File: Classes/Issues/GithubClient+Issues.swift [hunk 1]
```diff
@@ -289,13 +289,21 @@ extension GithubClient {
             return
         }
 
-        // https://developer.github.com/v3/repos/collaborators/#check-if-a-user-is-a-collaborator
+        // https://developer.github.com/v3/repos/collaborators/#review-a-users-permission-level
         request(Request(
-            path: "repos/\(owner)/\(repo)/collaborators/\(viewer)",
+            path: "repos/\(owner)/\(repo)/collaborators/\(viewer)/permission",
             headers: ["Accept": "application/vnd.github.hellcat-preview+json"],
```
Comment (Sherlouk) [inline Classes/Issues/GithubClient+Issues.swift]: Does this endpoint still need the accepts value?

**Comment by Sherlouk** [comment]
Yaas!

File: Classes/Issues/GithubClient+Issues.swift [hunk 2]
```diff
@@ -289,13 +289,21 @@ extension GithubClient {
...
             completion: { (response, _) in
-                // documentation states that collab = 204
-                completion(.success(response.response?.statusCode == 204))
+                let statusCode = response.response?.statusCode
+                if statusCode == 200,
+                    let json = response.value as? [String: Any],
+                    let permission = json["permission"] as? String {
```
Comment (Sherlouk) [inline Classes/Issues/GithubClient+Issues.swift]: Super minor but is it worth loading these permissions into an enum and passing that value through to be handled else where?

The enum could then have a `canManageIssue: Bool` which returns true if `self == .admin || self == .write`.

Just thinking ahead we may want to use the user's permission for other things and makes it a bit more flexible without locking this down to strictly this one use case?

File: Classes/Issues/GithubClient+Issues.swift (hunk 1 above)
Comment (rnystrom) [reply Classes/Issues/GithubClient+Issues.swift]: Good catch, nope!

File: Classes/Issues/GithubClient+Issues.swift (hunk 2 above)
Comment (rnystrom) [reply Classes/Issues/GithubClient+Issues.swift]: I was wondering this too. Doesn't hurt either way. I'll add this!

File: Classes/Issues/GithubClient+Issues.swift [hunk 3]
```diff
@@ -279,23 +279,51 @@ extension GithubClient {
         }))
     }
 
+    enum CollaboratorPermission: String {
+        case admin = "admin"
+        case write = "write"
+        case read = "read"
```
Comment (Sherlouk) [inline Classes/Issues/GithubClient+Issues.swift]: Values are unnecessary here, defaults to same as name

Comment (rnystrom) [reply Classes/Issues/GithubClient+Issues.swift]: 😱 TIL!

Comment (BasThomas) [reply Classes/Issues/GithubClient+Issues.swift]: It's awesome, been around since Swift 2 I think!

**Pull request merged**